server/src/main/java/io/crate/replication/logical/repository/LogicalReplicationRepository.java

```diff
@@ -179,8 +200,8 @@ public CompletableFuture<Collection<IndexMetadata>> getSnapshotIndexMetadata(Rep
                                                                                  SnapshotId snapshotId,
                                                                                  Collection<IndexId> indexIds) {
         assert SNAPSHOT_ID.equals(snapshotId) : "SubscriptionRepository only supports " + SNAPSHOT_ID + " as the SnapshotId";
-        var remoteIndices = indexIds.stream().map(IndexId::getName).toArray(String[]::new);
-        return getRemoteClusterState(remoteIndices).thenApply(response -> {
+        var remoteRelationNames = indexIds.stream().map(indexId -> IndexName.decode(indexId.getName()).toRelationName()).toArray(RelationName[]::new);
+        return getRemoteClusterState(remoteRelationNames).thenApply(response -> {
```

Comment:
I'm not sure if this will work out once we move away from index names that match the relation name. 

We'd have to map to the `relationNames` based on `RelationMetadata.Table#indexUUIDs` or something like that?